File: firebird/clojure/read/firebird_read.clj
; -----------------------------------------------------------------
;
;	firebird_read.clj
;
;					Feb/17/2011
;
; -----------------------------------------------------------------
(load-file "/var/www/data_base/common/clojure_common/sql_manipulate.clj")

; -----------------------------------------------------------------
(use 'clojure.contrib.sql)

(println "*** 開始 ***")
(let [db-host "localhost"
      db-name "/var/tmp/firebird/cities.fdb"]

  (def db {:classname "org.firebirdsql.jdbc.FBDriver" ; must be in classpath
           :subprotocol "firebirdsql"
           :subname (str "" db-host ":" db-name)
           ; Any additional keys are passed to the driver
           ; as driver-specific properties.
           :user "sysdba"
           :password "tiger"})

	(with-connection db
		(with-query-results rs ["select * from cities"]
			(display_proc rs)
)))
(println "*** 終了 ***")
;;      (dorun (map #(println (:language :iso_code %)) rs))))) 
; -----------------------------------------------------------------

File: firebird/rhino/read/firebird_read.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// ------------------------------------------------------------------
//
//	firebird_read.js
//
//						Nov/11/2010
//
// ------------------------------------------------------------------
importPackage(java.lang);
importPackage(java.sql);
load ("/var/www/data_base/common/rhino_common/sql_manipulate.js");

print	("*** 開始 ***");

Class.forName ("org.firebirdsql.jdbc.FBDriver");

var user = "sysdba";
var passwd = "tiger";
var database = "/var/tmp/firebird/cities.fdb";

var url = "jdbc:firebirdsql:localhost:" + database;

var conn = DriverManager.getConnection (url,user,passwd);

disp_proc (conn);

conn.close();

print	("*** 終了 ***");
// ------------------------------------------------------------------
